packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2745,75 +2755,112 @@ class RenderMouseRegion extends RenderProxyBox {
   @visibleForTesting
   MouseTrackerAnnotation get hoverAnnotation => _hoverAnnotation;
 
-  void _updateAnnotations() {
+  MouseTrackerAnnotation _buildAnnotation() {
+    if ((_onEnter != null ||
+         _onHover != null ||
+         _onExit != null ||
+         opaque
+        ) &&
+        RendererBinding.instance.mouseTracker.mouseIsConnected) {
+      return MouseTrackerAnnotation(
+        onEnter: _handleEnter,
+        onHover: _handleHover,
+        onExit: _handleExit,
+        key: _annotationKey,
+      );
+    }
+    return null;
+  }
+
+  // About updating the properties of [RenderMouseRegion]:
+  //
+  // `RenderMouseRegion` has 2 categories of properties:
+  //
+  //  * "Loose properties": [RenderMouseRegion] cares if these properties are
+  //    effective (an effective property makes _annotationIsActive true), but
+  //    doesn't care about their exact values. Examples are all the callbacks;
+  //    their actual values are not important since they are proxied. If their
+  //    changes don't affect `_annotationIsActive`, `markNeedsPaint` can be
+  //    skipped.
+  //  * "Strict properties": [RenderMouseRegion] needs the exact values of these
+  //    properties. An example is `opaque`. Their changes must be followed by
+  //    `markNeedsPaint` in order to update their exact values to the tree.
+
+  void _markPropertyUpdated({@required bool strict}) {
+    if (strict)
+      _markAnnotationDirty();
+    _updateAnnotationActiveBit();
+  }
+
+  // Call this method when a property has changed and might affect the
+  // `_annotationIsActive` bit.
+  //
+  // This method does NOT call `markNeedsPaint` unless the `_annotationIsActive`
+  // bit is changed. If there is a property that needs updating while
+  // `_annotationIsActive` stays true, call `_markAnnotationDirty`.
+  //
+  // This method must not be called during `paint`.
+  void _updateAnnotationActiveBit() {
```

Comment:
Maybe `_updateAnnotationIsActive`?

Reply:
but actually i guess what it really does is update the hover annotation, so maybe `_updateHoverAnnotation`?